File: hellolist/components/root/android/root.kt
package org.opengamestudio

typealias RC = RootContext

object RootComponent {
    init {
        val vm = VM
        val oneliners = arrayOf(
            F.isVisible, { c: RC -> vm.rootIsVisible.value = c.isVisible },
            F.items, { c: RC -> rootResetItems(vm, c.items) },
        )
        registerOneliners(rootCtrl(), oneliners)
    }

    fun setup() {
        rootSet(F.didSetup, true)
    }
}

File: hellolist/components/root/ios/root.swift
import hl

typealias RC = RootContext

class RootComponent {
    init() {
        var vm = { VM.singleton! }

        let effects: [Any] = [
            F.isVisible, { (c: RC) in vm().rootIsVisible = c.isVisible },
            F.items, { (c: RC) in rootResetItems(vm(), c.items) },
        ]
        var r: RC? = registerOneliners(rootCtrl(), effects)
    }

    func setup() {
        rootSet(F.didSetup, true)
    }
}
